staticjinja/reloader.py

```diff
@@ -39,16 +83,45 @@ def event_handler(self, event_type, src_path):
 
         """
         filename = os.path.relpath(src_path, self.searchpath)
+        # Avoid vim super temporary file
+        if filename.endswith('4913'):
+            return
         if self.should_handle(event_type, src_path):
-            print("%s %s" % (event_type, filename))
-            if self.site.is_static(filename):
-                files = self.site.get_dependencies(filename)
-                self.site.copy_static(files)
-            else:
-                templates = self.site.get_dependencies(filename)
-                self.site.render_templates(templates)
+            self.logger.info("%s %s" % (event_type, filename))
+            if event_type in [CREATED, MODIFIED]:
+                # Here we suspect some editing gets incorrectly reported as a
+                # creation so we include the event type created
+                try:
+                    source = self.sources.source_from_name(filename)
+                except SourceNotFoundError:
+                    flavor = self.sources.classify(
+                            filename,
+                            datapaths=self.datapaths,
+                            staticpaths=self.staticpaths)
+
+                    source = Source(
+                                filename,
+                                flavor,
+                                self.env,
+                                None
+                                )
+                    self.sources.add_source(source)
+                else:
+                    self.sources.update_dep_graph(source)
+                deps = self.sources.get_dependencies(source)
+                self.builder.handle_sources(deps)
+            else:  # event_type == DELETED:
+                # Here again we must suspect strange phenomena with temporary
+                # files whose deletion is detected by watchdog but not their
+                # creation (vim backup files for instance...)
+                try:
+                    source = self.sources.source_from_name(filename)
+                except SourceNotFoundError:
+                    return
+                else:
+                    self.sources.remove_source(source)
 
     def watch(self):
         """Watch and reload modified templates."""
-        import easywatch
-        easywatch.watch(self.searchpath, self.event_handler)
+        from .easywatch import watch
```

Comment:
We should move this line to the top of the file. The reason I placed it here originally was so that users could install staticjinja without easywatch if they didn't want it. Had I placed it at the top at they didn't have easywatch they would have got an `ImportError`.